### ISSUE TITLE
Question consering international US keyboard issues

### ISSUE THREAD

**Issue by naatje80** [TurboVNC/turbovnc#215]
Although not urgent, we get multiple questions concerning the behavior of the turbovnc viewer, when the default Keyboard in Windows 10 is setup as US International. I was curious if the behavior we notice is normal or could be considered as a bug. 

Although the behavior almost works similar to the commercial VNC software we used until now, we noticed one minor difference which results in feedback from our users that there "modifier keys" do not seem to respond.

I'll try to describe the issue:
When a US international keyboard is used, for example pressing the double quotes together with for the `a` charter, will result in: `ä`, Normally, if there is no modifier available for the specific second character, the output should just be the double quotes plus the character (at least for the commercial VNC, but most of our applications respond the same). For example pressing double quote followed by the `c`, should result in: `"c`. However, this does not work. To achieve this, we always have to type: `"` followed by `space `followed by `c`. 

Within a terminal (although I did not yet found a procedure that is 100% reproduceable), we noticed that, although e.g. the `"` followed by `c` does not generate something on screen during typing, a non-ascii character could be generated. We notice this as soon as we try to execute the command and also look trough the history of previous commands. 
An example of my history with this behavior:

![keyboard_issue](https://user-images.githubusercontent.com/26620355/78755009-bb156780-7978-11ea-8018-179fd340dcc6.png)

As soon as found a procedure that is always reproducible, I will share this with you.

**Comment by naatje80** [comment]
If found something this is always reproducable:
Type:
`echo "a`
This should show up in the terminal as:
`echo ä`
Remove the `ä` character, and type something followed by enter:
`echo something`
This results in:
`??something`

**Comment by dcommander** [comment]
Probably a duplicate of https://github.com/TurboVNC/turbovnc/issues/194

**Comment by naatje80** [comment]
@dcommander Thank you in advance. In our case we are not referring to the java vncviewer, but we see this behavior in the "normal" Windows vncviewer.

**Comment by dcommander** [comment]
I’ll see if I can reproduce it.

**Comment by naatje80** [comment]
Unfortunately this issue seems to bigger. Most likely the different behavior is also causing Pycharm to not except double quote characters for example. Switching to the standard US keyboard also solves the issue for Pycharm.

**Comment by dcommander** [comment]
I cannot reproduce the issue.  I tried reproducing it as you described in the original post, by typing `"c`.  As you noticed, that does not produce `"c` in a TurboVNC session, but on my Linux machine, it doesn't produce `"c` on the local display either (also when using a U.S. Intl keyboard layout.)  As near as I can figure, TurboVNC is properly emulating the behavior of the local display in that regard.

I was not able to reproduce the issue using the procedure you described in https://github.com/TurboVNC/turbovnc/issues/215#issuecomment-610803439 at all.

Perhaps this has something to do with your server-side window manager or something else that is different in your environment.  The more details you provide, the better chance I have of reproducing it.

**Comment by dcommander** [comment]
@naatje80 Please provide more information about your environment or any other information that might help in reproducing the issue.

**Comment by naatje80** [comment]
@dcommander I'm sorry for my delayed response, most of us had a prolonged weekend in our country due to a public holiday.  But I indeed will further investigate what could be the cause of our issue. I did not mention we are using TigerVNC for the server, so I will also check if the combination with the TurboVNC server is also causing issues in our case. I will give you a more detailed feedback to help you reproduce the issue.

**Comment by naatje80** [comment]
@dcommander We also got the same issue with the turbovnc server. But I found out that my described issues were in fact three different issues/unclarities.

1. The issue with the strange characters in combination with the international keyboard (which I described here: https://github.com/TurboVNC/turbovnc/issues/215#issuecomment-610803439), only seems to appear when the LC_CTYPE is not set to en_US.UTF-8. When we unset LC_ALL and change the LC_CTYPE to the default en_US.UTF-8 value the issue is resolved. I hope this helps to reproduce this issue.
2. The Pycharm issue seems to be a combination of my first point together with this issue: https://bugs.centos.org/view.php?id=16545. We encountered similar issues with another commercial product (which was not related to VNC), which we resolved by overruling the following environment variables in the wrapper script of the application: 
`export QT_IM_MODULE=xim
export XMODIFIERS=@im=none
`. The issue with pycharm is, most likely, also fixed if the issue of the first point is resolved (in combination with a change from our side to overrule the above environment variables).
3. The different behavior for the international keyboard is indeed the same on a normal Centos/Linux installation. I was confused by the fact that the commercial VNC product copied the "normal" Windows behavior instead of the way the Gnome Shell desktop normal works with an international keyboard.

To summarize: I think the only issue from the turbovnc viewer side is the issue with the unreadable characters that are created when the LC_CTYPE is set to non en_US.UTF8 in combination with a the US International Keyboard.

**Comment by dcommander** [comment]
I can reproduce (1), but it also occurs on the local display without TurboVNC, so again, I don't think it's a TurboVNC bug.

**Comment by dcommander** [comment]
BTW, you can say "RealVNC."  Everyone knows that that's the only commercial Linux VNC product out there.  ;)

**Comment by dcommander** [comment]
Closing, but if you feel that my analysis is incorrect, then please feel free to comment further.

**Comment by naatje80** [comment]
You are again correct: this is not an issue with TurboVNC. I was again mislead by the different behavior of RealVNC :) . I noticed that for our RealVNC sessions the LC_CTYPE is indeed set to en_US.UTF-8. I will check if our tigervnc server skips a certain configuration file or if RealVNC forces the different character encoding set. Otherwise I will ask our system management if we really still need the non-standard encoding or if this is just pure history of old settings that are still implemented.

**Comment by naatje80** [comment]
It seems that this was also an issue in RealVNC. It seems I already changed this for my environment a few years ago, which I already forgot. So I will contact our system manager and discuss with him if we still need the C value for the locale settings :). Thank you again for all your help.

**Comment by naatje80** [comment]
One more update: Only changing the LC_CTYPE to en_US.UTF-8 also solves the issue for Pycharm.
